homeassistant/components/songpal/coordinator.py

```diff
@@ -0,0 +1,259 @@
+"""Support for Songpal-enabled (Sony) media devices."""
+
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+import logging
+from typing import Any
+
+from songpal import (
+    ConnectChange,
+    ContentChange,
+    Device,
+    PowerChange,
+    SettingChange,
+    SongpalException,
+    VolumeChange,
+)
+from songpal.containers import Setting
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_ENDPOINT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PARAM_NAME = "name"
+PARAM_VALUE = "value"
+
+INITIAL_RETRY_DELAY = 10
+
+type SongpalConfigEntry = ConfigEntry[SongpalCoordinator]
+
+
+class SongpalCoordinator(DataUpdateCoordinator[dict[str, Any]]):
```

Comment:
Can we create a data class for the return type? This way we can make things typed and have more checks